src/vcpkg/base/cofffilereader.cpp

```diff
@@ -111,19 +111,25 @@ namespace vcpkg
 
         char file_start[FILE_START.size()];
         Checks::check_exit(VCPKG_LINE_INFO, fs.read(&file_start, FILE_START.size(), 1) == 1);
-        Checks::check_exit(VCPKG_LINE_INFO,
-                           FILE_START == StringView{file_start, FILE_START.size()},
-                           "Incorrect archive file signature");
+
+        if (FILE_START != StringView{file_start, FILE_START.size()})
+        {
+            msg::println_error(msgIncorrectArchiveFileSignature);
+            Checks::exit_fail(VCPKG_LINE_INFO);
+        }
     }
 
     static void read_and_skip_first_linker_member(const vcpkg::ReadFilePointer& fs)
     {
         ArchiveMemberHeader first_linker_member_header;
         Checks::check_exit(VCPKG_LINE_INFO,
                            fs.read(&first_linker_member_header, sizeof(first_linker_member_header), 1) == 1);
-        Checks::check_exit(VCPKG_LINE_INFO,
-                           memcmp(first_linker_member_header.name, "/ ", 2) == 0,
-                           "Could not find proper first linker member");
+        if (memcmp(first_linker_member_header.name, "/ ", 2) != 0)
+        {
+            Debug::println("Could not find proper first linker member");
+            Checks::exit_fail(VCPKG_LINE_INFO);
+        }
+
```

Comment:
Why? Now vcpkg simply exists without any error message. For example:
```
...
-- Installing x64-mingw-static-rel
-- Fixing pkgconfig file: G:/vcpkg4/packages/jemalloc_x64-mingw-static/lib/pkgconfig/jemalloc.pc
-- Fixing pkgconfig file: G:/vcpkg4/packages/jemalloc_x64-mingw-static/debug/lib/pkgconfig/jemalloc.pc
-- Installing: G:/vcpkg4/packages/jemalloc_x64-mingw-static/share/jemalloc/copyright
-- Performing post-build validation
admin@DESKTOP-NDEQJD0 MSYS /g/vcpkg4 (master)
$
```
Instead of showing an error message like before 
```